xds/internal/xdsclient/transport/transport.go

```diff
@@ -193,18 +193,13 @@ func New(opts Options) (*Transport, error) {
 	}
 
 	// Dial the xDS management with the passed in credentials.
-	dopts := []grpc.DialOption{
-		opts.ServerCfg.CredsDialOption(),
+	dopts := append(opts.ServerCfg.DialOptions(),
 		grpc.WithKeepaliveParams(keepalive.ClientParameters{
 			// We decided to use these sane defaults in all languages, and
 			// kicked the can down the road as far making these configurable.
 			Time:    5 * time.Minute,
 			Timeout: 20 * time.Second,
-		}),
-	}
-	if dialerOpts := opts.ServerCfg.DialerOption(); dialerOpts != nil {
-		dopts = append(dopts, dialerOpts)
-	}
+		}))
```

Comment:
While you are here, it might make sense to refactor this to be as follows:

```golang
kpCfg := grpc.WithKeepaliveParams(keepalive.ClientParameters{
	// We decided to use these sane defaults in all languages, and
	// kicked the can down the road as far making these configurable.
	Time:    5 * time.Minute,
	Timeout: 20 * time.Second,
}))
dopts := append(opts.ServerCfg.DialOptions(), kpCfg)
```

Reply:
Hmm .. actually thinking about it a little more. I think this might be better:

```golang
	// Dial the xDS management server with dial options specified by the server
	// configuration and a static keepalive configuration that is common across
	// gRPC language implementations.
	kpCfg := grpc.WithKeepaliveParams(keepalive.ClientParameters{
		Time:    5 * time.Minute,
		Timeout: 20 * time.Second,
	})
	dopts := append([]grpc.DialOption{kpCfg}, opts.ServerCfg.DialOptions()...)
```

You might find the following blogs interesting to read: https://go.dev/blog/slices, https://go.dev/blog/slices-intro

Reply:
Done